src/dtgtk/thumbtable.c

```diff
@@ -1481,38 +1481,53 @@ static void _event_dnd_begin(GtkWidget *widget, GdkDragContext *context, gpointe
 
   table->drag_list = g_list_copy((GList *)dt_view_get_images_to_act_on(FALSE, TRUE));
 
-  // if we are dragging a single image -> use the thumbnail of that image
-  // otherwise use the generic d&d icon
-  // TODO: have something pretty in the 2nd case, too.
-  if(g_list_length(table->drag_list) == 1)
+#ifdef HAVE_MAP
+  dt_view_manager_t *vm = darktable.view_manager;
+  dt_view_t *view = vm->current_view;
+  if(!strcmp(view->module_name, "map"))
   {
-    const int id = GPOINTER_TO_INT(g_list_nth_data(table->drag_list, 0));
-    dt_mipmap_buffer_t buf;
-    dt_mipmap_size_t mip = dt_mipmap_cache_get_matching_size(darktable.mipmap_cache, ts, ts);
-    dt_mipmap_cache_get(darktable.mipmap_cache, &buf, id, mip, DT_MIPMAP_BLOCKING, 'r');
-
-    if(buf.buf)
+    if(table->drag_list)
+      dt_view_map_drag_set_icon(darktable.view_manager, context,
+                                GPOINTER_TO_INT(table->drag_list->data),
+                                g_list_length(table->drag_list));
+  }
+  else
+  {
+#endif
```

Comment:
I suppose this can be moved above...

Reply:
Sorry, I don't get you. Could you tell me a bit more ?
If HAVE_MAP is false, the code must be as before. But if TRUE, the `if{ new code }else{ old code }` must be complete.

Reply:
In C a simple  block like this is correct:
```
{
   ...
}
```

So I propose to use a single HAVE_MAP:

```
#if HAVE_MAP
if (...)
{
   ...
}
else
#endif
{
   ...
}
```

Reply:
so obvious ! thanks for your patience !

Reply:
done